### DISCUSSION TITLE
Calculating S to P Receiver Function using Local Network Data

### DISCUSSION THREAD

**Discussion by geoAP90** [PyGLImER/PyGLImER#140] [Q&A]
### Discussed in https://github.com/PyGLImER/PyGLImER/discussions/138

<div type='discussions-op-text'>

<sup>Originally posted by **geoAP90** March 14, 2023</sup>
Hello, 
 I am trying to fit my local network data in PyGLimmer for the computation of S to P RF.
I have some questions regarding the preprocessing steps. 
1. def yield_st():
    static = os.path.join(
        db_base_path, 'static_data', 'database_sac', 'waveforms', 'local_db')
    networks = ['IU']
    stations = ['HRV']
    for net, stat in zip(networks, stations):
        st = read(os.path.join(static, net, stat, 'hrv10.mseed'))
        yield st
Here, you have mentioned 'hrv10.mseed' as your input file. My data is in mseed format but there are 3(N, Z, E) mseed files for each event. How should because there was only one event. How should I make my input database for PyGLimmer?

Thank you for your patience and time.

</div>

**Comment by PeterMakus** [comment]
Hi Arpita,
In your case, it might be a good idea to go over some basic concepts in the usage of obspy - for example, by having a look at their excellent [tutorial](https://docs.obspy.org/tutorial/index.html). PyGLImER relies heavily on obspy and it will probably be challenging to understand the code's logic & API without having some experience with obspy.
In the given case, the ``obspy.core.Stream`` object is of particular relevance. Since you have three files, you could do something along the lines of:

```python
from obspy import read
# ...
st = read('path_to_your_first_file.mseed')
st.extend(read('path_to_your_second_file.mseed')
# and so on
# ...
```

**Reply by geoAP90** [reply]
![Screenshot from 2023-03-15 11-49-23](https://user-images.githubusercontent.com/37789488/225223605-26eec85c-23c1-4d38-a203-ea303ddc3b86.png) have merged the three channels data for each day in one  .mseed file. I am getting this "2023-03-15 11:01:13,312 - DEBUG - No valid arrival found for station OBS01,event smi:service.iris.edu/fdsnws/event/1/query?eventid=11172804, and phase P" this message as I run this following line "R.import_database(yield_st, yield_inv)"

I have attached a screenshot of the obspy function where I have read the waveform data

I have now my station.xml, event catalogue and  waveform data in the asked format. But, why it is not able to preprocess my raw data?

**Reply by PeterMakus** [reply]
The error message basically tells you that there are no teleseismic S-wave arrivals within the usable epicentral distance for the given event. Judging from the attached plot, there are no arrivals recorded at your station also.

**Reply by geoAP90** [reply]
Ok Thank you. I get that. I will try with different event

**Comment by lsawade** [comment]
To elaborate on what Peter was saying: The event id you provided ([link to XML](service.iris.edu/fdsnws/event/1/query?eventid=11172804)) is for an event that occurred at `2020-01-18T16:38:14.384`. If you compare that to your date range for your traces, you can see that the event cannot be in your traces.

Another question, your trace components show (X,Y,Z). Are you certain they are oriented in (N,E,Z)? Or do you need to rotate them. This can be a problem for the deconvolution later on, as we (you, and the code) need to rotated traces to (R,T,Z) to properly compute receiver functions.
